src/java/org/apache/cassandra/cql3/UpdateParameters.java

```diff
@@ -163,8 +166,21 @@ public Cell<?> addCell(ColumnMetadata column, ByteBuffer value) throws InvalidRe
 
     public Cell<?> addCell(ColumnMetadata column, CellPath path, ByteBuffer value) throws InvalidRequestException
     {
+        // General column value size
         Guardrails.columnValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
 
+        if (column.type instanceof AsciiType) // Ascii size specific guardrail
+        {
+            Guardrails.columnAsciiValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
+        } else if (column.type instanceof BytesType) // Blob size specific guardrail
+        {
+            Guardrails.columnBlobValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
+        } else if (column.type instanceof UTF8Type) // text and varchar size specific guardrails
+        {
+            Guardrails.columnTextValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
+            Guardrails.columnVarcharValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
+        }
```

Comment:
It is confusing (sometimes surprising) to end-users to have 2 guardrails to check on the same value. Since `text` and `varchar` are identical internally, can you combine those 2 guardrails?